### PR TITLE
Hugbox megafauna logging

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Tells you the name of the person the boss was last chasing when they teleported

I guess this might miss things like trying to send the shuttle, being gibbed, and then having it launched though. Not sure how to best improve it.

### CODE REVIEW

**Comment by Saegrimr** [comment]
Just need who pressed the button. There's already an alert for "Monster on its way", so we're already halfway there.

File: code/game/objects/effects/portals.dm [hunk 1]
```diff
@@ -62,7 +62,12 @@
 		return
 	if (istype(M, /atom/movable))
 		if(istype(M, /mob/living/simple_animal/hostile/megafauna))
-			message_admins("[M] (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[M]'>FLW</A>) has teleported through [src].")
+			var/mob/living/T
+			var/mob/living/simple_animal/hostile/H = M
+			if(H.target)
+				T = H.target
+			message_admins("[M] (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[M]'>FLW</A>) has teleported through [src]. Last targeting: [key_name_admin(T)]")
```
Comment (coiax) [inline code/game/objects/effects/portals.dm]: Use the new admin message defines please

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 2]
```diff
@@ -40,10 +40,13 @@
...
 		return
 	var/turf/newloc = loc
+	var/mob/living/T
+	if(target)
+		T = target
 	message_admins("Megafauna [src] \
 		(<A HREF='?_src_=holder;adminplayerobservefollow=\ref[src]'>FLW</A>) \
 		moved via shuttle from ([oldloc.x],[oldloc.y],[oldloc.z]) to \
-		([newloc.x],[newloc.y],[newloc.z])")
+		([newloc.x],[newloc.y],[newloc.z]) Last targeting: [key_name_admin(T)]")
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Use the new admin message defines please

**Comment by KorPhaeron** [comment]
Pressing the button won't fix the portal thing though.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: what are they

Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/admin.dm#L40

**Comment by Shadowlight213** [comment]
@KorPhaeron we also need logging on who pressed the button to send the shuttle.

**Comment by KorPhaeron** [comment]
well I'm going away until next tuesday again so someone else should probably finish this

**Comment by KorPhaeron** [comment]
I will reopen later to avoid triggering oranges

**Pull request closed without merging**